### DISCUSSION TITLE
What does XamlCheckProcessRequirements do?

### DISCUSSION THREAD

**Discussion by Balkoth** [microsoft/WindowsAppSDK#2872] [Q&A] [answered]
I can not find any documentation on what it does nor if or why it is needed. The application seems to run even if you omit the call. So what gives?

**Comment by BenJKuhn** [comment, marked as answer]
It actually works the other way around. It ensures that the process can run XAML, and provides a deterministic error if a check fails. Otherwise, it quietly does nothing. 

This was originally introduced to ensure that attempting to run Windows App SDK from an elevated process gave a "nice" error until we could provide full support, at which point the check was removed. Without this check, the app would fail during WinRT type lookup, leading to a "type not found" error that would be difficult to root-cause. 

Now that elevated processes are supported, the check is a no-op but the function is retained for binary compatibility.

- Ben

**Reply by Balkoth** [reply]
So is this an implementation detail or can this be documented somewhere? I am asking, because if you want to implement the program class for yourself, this is something you step over.

**Reply by BenJKuhn** [reply]
It's an implementation detail. I'm not aware of any plans to make it do anything useful anytime soon, but as long as XAML is putting it in the generated code, it would be a best practice to mirror that if you're hand-rolling just to be consistent.

**Reply by Balkoth** [reply]
Thanks for the heads up. Would be best if you did not ever need to modify this.
